mipengine/controller/node_landscape_aggregator.py

```diff
@@ -263,44 +269,42 @@ async def _gather_all_dataset_infos(
 
     for node_info in nodes:
         node_socket_addr = _get_node_socket_addr(node_info)
-        datasets_per_data_model = await _get_node_datasets_per_data_model(
-            node_socket_addr
-        )
-        for data_model, datasets in datasets_per_data_model.items():
-            current_labels = (
-                aggregated_datasets[data_model]
-                if data_model in aggregated_datasets
-                else {}
-            )
-            current_datasets = (
-                dataset_locations[data_model] if data_model in dataset_locations else {}
-            )
-
-            for dataset in datasets:
-                current_labels[dataset] = datasets[dataset]
+        datasets_per_data_model = _get_node_datasets_per_data_model(node_socket_addr)
+        if datasets_per_data_model:
```

Comment:
nit: This if statement is not needed if the `_get_node_datasets_per_data_model` returns empty dict instead of None.